### ISSUE TITLE
Man page sources are opened as 'nomodifiable' by default

### ISSUE THREAD

**Issue by kdkasad** [neovim/neovim#13139]
- `nvim --version`: **v0.4.4 (Release)**
- `vim -u DEFAULTS` (version: ) behaves differently? **Yes**
- Operating system/version: **Artix Linux**
- Terminal name/version: **Alacritty 0.5.0**
- `$TERM`: **xterm-256color**

### Steps to reproduce using `nvim -u NORC`

Doesn't reproduce with `-u NORC` because it uses `$VIM_RUNTIME/ftplugin/man.vim`. This file is provided with neovim, and is run when the filetype is detected as `man`. You can reproduce with:
```
nvim <man page source file>
```
I encountered this when editing the manpage for `kirc(1)`, [available here](https://git.kasad.com/kirc.git/plain/kirc.1).

### Actual behaviour
When editing a man page source, it is opened with the following options set:
```
set readonly buftype=nofile nomodifiable
```
This would make sense if the manpage content was rendered, because then it wouldn't be editing the actual file. However, the unchanged file contents are displayed in the editor and the file is still not editable.

### Expected behaviour
Neovim should open man page sources as normal (editable) files.
The exception to this would be if it is rendering their content into some other form.

### Extra information
I've tracked the problem to the file `$VIM_RUNTIME/ftplugin/man.vim`, which is installed as `/usr/share/nvim/runtime/ftplugin/man.vim` on my system.

The following lines from the file are responsible:
```
setlocal noswapfile buftype=nofile bufhidden=hide
setlocal nomodified readonly nomodifiable
```
I know that removing these solves the problem, but I'm reporting this as a bug because the default behavior (which includes these lines) seems unreasonable.

**Comment by glacambre** [comment]
The problem is that man source files have the same filetype as `:Man` buffers. @nhooyr I think you are the `man.vim` expert, do you have ideas about the best way to fix this? :)

**Comment by justinmk** [comment]
fixed by #15488 , which is less loose about when to invoke "pager mode".
